ui/src/app/machines/views/MachineDetails/MachineNetwork/EditInterface/EditInterface.test.tsx

```diff
@@ -71,6 +72,61 @@ describe("EditInterface", () => {
       </Provider>
     );
     expect(wrapper.find("EditPhysicalForm").exists()).toBe(true);
-    expect(wrapper.find("FormCard").prop("title")).toBe("Edit physical");
+    expect(wrapper.find("FormCard").prop("title")).toBe("Edit Physical");
+  });
+
+  it("displays a form for editing an alias", () => {
+    const link = networkLinkFactory();
+    const nic = machineInterfaceFactory({
+      links: [networkLinkFactory(), link],
+      type: NetworkInterfaceTypes.PHYSICAL,
+    });
+    state.machine.items = [
+      machineDetailsFactory({
+        system_id: "abc123",
+        interfaces: [nic],
+      }),
+    ];
+    const store = mockStore(state);
+    const wrapper = mount(
+      <Provider store={store}>
+        <MemoryRouter
+          initialEntries={[{ pathname: "/machines", key: "testKey" }]}
+        >
+          <EditInterface
+            systemId="abc123"
+            close={jest.fn()}
+            linkId={link.id}
+            nicId={nic.id}
+          />
+        </MemoryRouter>
+      </Provider>
+    );
+    expect(wrapper.find("EditAliasOrVlanForm").exists()).toBe(true);
+    expect(wrapper.find("FormCard").prop("title")).toBe("Edit Alias");
+  });
+
+  it("displays a form for editing a VLAN", () => {
+    const nic = machineInterfaceFactory({
+      type: NetworkInterfaceTypes.VLAN,
+    });
+    state.machine.items = [
+      machineDetailsFactory({
+        system_id: "abc123",
+        interfaces: [nic],
+      }),
+    ];
+    const store = mockStore(state);
+    const wrapper = mount(
+      <Provider store={store}>
+        <MemoryRouter
+          initialEntries={[{ pathname: "/machines", key: "testKey" }]}
+        >
+          <EditInterface systemId="abc123" close={jest.fn()} nicId={nic.id} />
+        </MemoryRouter>
+      </Provider>
+    );
+    expect(wrapper.find("EditAliasOrVlanForm").exists()).toBe(true);
+    expect(wrapper.find("FormCard").prop("title")).toBe("Edit VLAN");
```

Comment:
Nice way to detect which one is being displayed but slightly coupled to copy changes :man_shrugging:

Reply:
I think this might be OK as we do want to check that we're updating the title with the type of the interface, I guess it could just check for `VLAN` in the string...

Reply:
Not sure its worth it in this case but something to keep in mind.